query/javascript/src/Query.ts

```diff
@@ -157,4 +167,51 @@ export default class Query {
     | undefined {
     return this.stepMatchArgumentsListsByPickleStepId.get(pickleStepId)
   }
+
+  public getBeforeHookSteps(
+    pickleId: string
+  ): ReadonlyArray<messages.TestCase.ITestStep> {
+    const testCase = this.testCaseByPickleId.get(pickleId)
+
+    // const hookSteps = testCase.testSteps.filter((step) => step.hookId)
+    const hookSteps: messages.TestCase.ITestStep[] = []
+    let pickleStepFound = false
+
+    for (const step of testCase.testSteps) {
+      if (!step.hookId) {
+        pickleStepFound = true
+      }
+      if (step.hookId && !pickleStepFound) {
+        hookSteps.push(step)
+      }
+    }
+
+    return hookSteps
+  }
+
+  public getAfterHookSteps(
```

Comment:
Can you extract the duplicated code between `getAfterHookSteps` and `getBeforeHookSteps` to a separate function?